src/fit.jl

```diff
@@ -89,41 +89,47 @@ end
 
 
 function train!(
-    estimator::LGBMEstimator, num_iterations::Int, tests_names::Vector{String}, verbosity::Integer, start_time::DateTime
+    estimator::LGBMEstimator, 
+    num_iterations::Int, 
+    tests_names::Vector{String}, 
+    verbosity::Integer, 
+    start_time::DateTime
 )
     results = Dict{String,Dict{String,Vector{Float64}}}()
-    n_tests = length(tests_names)
     metrics = LGBM_BoosterGetEvalNames(estimator.booster)
-    n_metrics = length(metrics)
-    bigger_is_better = [ifelse(in(metric, MAXIMIZE_METRICS), 1., -1.) for metric in metrics]
-    best_score = fill(-Inf, (n_metrics, n_tests))
-    best_iter = fill(1, (n_metrics, n_tests))
+
+    bigger_is_better = Dict(metric => ifelse(in(metric, MAXIMIZE_METRICS), 1., -1.) for metric in metrics)
+    best_score = Dict{String,Dict{String,Real}}()
+    best_iter = Dict{String,Dict{String,Real}}()
+
+    for metric in metrics
+        best_score[metric] = Dict{String,Real}()
+        best_iter[metric] = Dict{String,Real}()
+        for tests_name in tests_names
+            best_score[metric][tests_name] = -Inf
+            best_iter[metric][tests_name] = 1
+        end
+    end
 
     start_iter = get_iter_number(estimator) + 1
     end_iter = start_iter + num_iterations - 1
 
     metrics_idx_sequence = (((start_iter:end_iter) .- 1) .% estimator.metric_freq) .== 0
     total_metrics_evals = sum(metrics_idx_sequence)
```

Comment:
Looks like this one can be deleted?

Reply:
Yes! Thanks! (I was aiming to refactor the code so that I could get rid of it...but forgot to do so after all the refactoring!) :D